### ISSUE TITLE
Allow page title in breadcrumb to be rendered as h1

### ISSUE THREAD

**Issue by jpesce** [vtex-apps/store-discussion#650]
**What problem are you trying to solve?**
Having an `h1` in every page is a good SEO practice and screen real estate is a valuable thing — especially above the fold where titles are usually placed. The breadcrumb is a component that already has the page title in it. A common practice to avoid having a dedicated place in the page for the title is to simply render the corresponding breadcrumb  item as an `h1`.

<img width="1552" alt="Screen Shot 2021-10-11 at 10 42 32" src="https://user-images.githubusercontent.com/1228352/136801143-4db9ef98-04a6-4ca6-ac43-9d67f146eabb.png">

**Describe the solution you'd like**
Add a prop (`renderPageTitleAsH1`) to the breadcrumb component that checks if the breadcrumb item is also the page title. If it is, render it as `h1` instead of simply a link. We already have a component to [get the page title from the breadcrumb](https://github.com/vtex-apps/search-result/blob/master/react/hooks/useSearchTitle.ts) so we just need to apply the same logic.

**Why do you think this feature should be available to all stores?**
I think the scenario I mentioned above is pretty common and this would allow for a better control over the component's behavior.

**Describe alternatives you've considered**
We tried to remove the last item of the breadcrumb (via CSS’s `:last-child`) and add the `search-title` component after it. The problem is that the last breadcrumb item is not necessarily the title, causing weird behaviors such as this:
<img width="312" alt="Screen Shot 2021-10-11 at 11 02 08" src="https://user-images.githubusercontent.com/1228352/136803542-b6bb3ace-bf26-49cb-afbe-f4072a8fbade.png">
[Link to the problematic page](https://www.oficinareserva.com/roupas/camisetas/branco?initialMap=c,c&initialQuery=roupas/camisetas&map=category-1,category-2,cor)

**Comment by iaronaraujo** [comment]
Hi @jpesce ! Thank you a lot for this proposal! I think that from a product point of view this makes a lot of sense, but I am wondering what kind of drawbacks could this solution of comparing with the page title have.  Could this affect performance?

**Comment by jpesce** [comment]
@iaronaraujo I don't think it would have performance issues compared to using the page title component because it would use the same logic. At worst, with the prop enabled, it would have the same performance as having a breadcrumb component + a page title component. With the prop disabled, it would have the same performance as the current breadcrumb. We’d be glad to start working on this PR if you guys approve the idea and, if any performance issues arise, we can discuss it further.

**Comment by igorbrasileiro** [comment]
Hello @jpesce, first of all, thank you for this proposal. Adding a heading level 1 tag may impact accessibility and SEO with the wrong usage. In the accessibility aspect, a page can't have two heading level 1 as far as I know, and in the SEO aspect, the search engines evaluate this tag and its content to rank the pages. So, knowing that this feature's solution should not add this new behavior by default but via component property. Does this explanation make sense for you?

**Comment by jpesce** [comment]
Hi @igorbrasileiro , thanks for the response!

> So, knowing that this feature's solution should not add this new behavior by default but via component property

Completely agree. Our proposal is to add the `renderPageTitleAsH1` prop, making it `false` by default. This means that it doesn't change anything for the people who already use the component unless they actively enable the new prop.

The recommendation is to enable the prop only if the page doesn't have an `h1` somewhere else. That is the case in the example screenshot.

With that in mind, can we proceed with the development?

**Comment by igorbrasileiro** [comment]
Hey @jpesce! Ok ok, you can implement it 😄

**Comment by jpesce** [comment]
No longer pursuing this—feel free to reopen and reassign if needed.
